pkg/config/schema/schema.go

```diff
@@ -0,0 +1,607 @@
+}
+
+type NetworkProxyKubelets struct {
+	ByHostname bool `yaml:"byHostname,omitempty" json:"byHostname,omitempty"`
+	ByIP       bool `yaml:"byIP,omitempty" json:"byIP,omitempty"`
+}
+
+type Plugins struct {
+	Image  string                 `yaml:"image,omitempty" json:"image,omitempty"`
+	Config map[string]interface{} `yaml:"config,omitempty" json:"config,omitempty"`
+	RBAC   *PluginsRBAC           `yaml:"rbac,omitempty" json:"rbac,omitempty"`
+}
+
+type PluginsRBAC struct {
+	Role        []rbacv1.PolicyRule `yaml:"role,omitempty" json:"role,omitempty"`
+	ClusterRole []rbacv1.PolicyRule `yaml:"clusterRole,omitempty" json:"clusterRole,omitempty"`
+}
+
+type Plugin struct {
+	Plugins `yaml:",inline" json:",inline"`
+
+	Version string `yaml:"version,omitempty" json:"version,omitempty"`
+}
+
+type ControlPlane struct {
+	Distro           Distributions                `yaml:"distro,omitempty" json:"distro,omitempty"`
+	BackingStore     BackingStore                 `yaml:"backingStore,omitempty" json:"backingStore,omitempty"`
+	HostPathMapper   HostPathMapper               `yaml:"hostPathMapper,omitempty" json:"hostPathMapper,omitempty"`
+	CoreDNS          CoreDNS                      `yaml:"coredns,omitempty" json:"coredns,omitempty"`
+	VirtualScheduler *Enablement                  `yaml:"virtualScheduler,omitempty" json:"virtualScheduler,omitempty" product:"pro"`
+	Proxy            *ControlPlaneProxy           `yaml:"proxy,omitempty" json:"proxy,omitempty"`
+	Service          ControlPlaneService          `yaml:"service,omitempty" json:"service,omitempty"`
+	Ingress          ControlPlaneIngress          `yaml:"ingress,omitempty" json:"ingress,omitempty"`
+	HighAvailability ControlPlaneHighAvailability `yaml:"highAvailability,omitempty" json:"highAvailability,omitempty"`
+	Advanced         ControlPlaneAdvanced         `yaml:"advanced,omitempty" json:"advanced,omitempty"`
+}
+
+type Distributions struct {
+	K3S      *DistributionsK3s    `yaml:"k3s,omitempty" json:"k3s,omitempty"`
+	K8S      *DistributionsK8s    `yaml:"k8s,omitempty" json:"k8s,omitempty"`
+	K0S      *corev1.Container    `yaml:"k0s,omitempty" json:"k0s,omitempty"`
+	Advanced DistributionAdvanced `yaml:"advanced,omitempty" json:"advanced,omitempty"`
+}
+
+type DistributionsK3s struct {
+	corev1.Container `yaml:",inline" json:",inline"`
+	Token            string `yaml:"token,omitempty" json:"token,omitempty"`
+}
+
+func (d *DistributionsK3s) SetDefaults() {
+	if defaults.CanUpdate(d.Container.Resources) {
+		d.Container.Resources = corev1.ResourceRequirements{
+			Limits: map[corev1.ResourceName]resource.Quantity{
+				"cpu":    {Format: "100m"},
+				"memory": {Format: "256Mi"},
+			},
+			Requests: map[corev1.ResourceName]resource.Quantity{
+				"cpu":    {Format: "100m"},
+				"memory": {Format: "256Mi"},
+			},
+		}
+	}
+}
+
+type DistributionsK8s struct {
+	APIServer         *corev1.Container `yaml:"apiServer,omitempty" json:"apiServer,omitempty"`
```

Comment:
this should not be a container, since it's included in the syncer container, there should only be an image here